hibernate-core/src/main/java/org/hibernate/hql/internal/ast/util/JoinProcessor.java

```diff
@@ -94,9 +105,84 @@ public static JoinType toHibernateJoinType(int astJoinType) {
 		}
 	}
 
+	private <T extends AST> List<T> findAllNodes(AST node, Class<T> clazz) {
+		ArrayList<T> result = new ArrayList<>();
+		findAllNodesCore( node, clazz, result );
+		return result;
+	}
+
+	private <T extends AST> void findAllNodesCore(AST node, Class<T> clazz, List<T> found) {
```

Comment:
Maybe `doFindAllNodes`? I don't think we use `Core` anywhere else.

Reply:
fixed